src/components/ha-label-badge.ts

```diff
@@ -25,11 +27,23 @@ class HaLabelBadge extends LitElement {
     return html`
       <div class="badge-container">
         <div class="label-badge" id="badge">
+          ${this.upperLabel
+            ? html`
+                <div
+                  class=${classMap({
+                    upperLabel: true,
+                    big: this.upperLabel.length > 5,
+                  })}
+                >
+                  <span>${this.upperLabel}</span>
```

Comment:
I think is just similar to the rest but is the span needed?

Reply:
Removing the span and moving the css to the div actually does make it look a bit different. I didn't want to bother with it for this and just use what is there.